src/autobind/parser.cr

```diff
@@ -7,9 +7,36 @@ module Autobind
     protected getter index : Clang::Index
     protected getter translation_unit : Clang::TranslationUnit
     getter output = ""
+    getter name = "LibC"
+    getter? module_name : String? = nil
 
     def libc_output
-      "lib LibC\n#{@output}end\n"
+      check(
+        if mod = module_name?
+          "module #{mod}\nlib #{name}\n#{@output}end\nend\n"
+        else
+          "lib LibC\n#{@output}end\n"
+        end
+      )
+    end
+
+    private def check(output)
+      # check formatting
+      formatted = Crystal.format output
+      # check syntax
+      Crystal::Parser.parse formatted
+      formatted
+    rescue err
+      if err.is_a? Crystal::SyntaxException
```

Comment:
You can restrict the error type with `rescue err : Crystal::SyntaxException`

Reply:
I thought I had read that that didn't work but now that I think about it I was thinking of Javascript, which has similar syntax that _only works sometimes_.